astropy/table/tests/test_row.py

```diff
@@ -355,20 +361,35 @@ def test_uint_indexing():
 
     Regression test for gh-7464.
     """
-    t = table.Table([[1.0, 2.0, 3.0]], names="a")
-    assert t["a"][1] == 2.0
-    assert t["a"][np.int_(1)] == 2.0
-    assert t["a"][np.uint(1)] == 2.0
-    assert t[np.uint(1)]["a"] == 2.0
-
-    trepr = [
+    row_index = dtype(1)
+    assert small_table["a"][row_index] == 2.0
+    assert small_table[row_index]["a"] == 2.0
+    assert repr(small_table[row_index]).splitlines() == [
         "<Row index=1>",
-        "   a   ",
-        "float64",
-        "-------",
-        "    2.0",
+        "   a     b  ",
+        "float64 bool",
+        "------- ----",
+        "    2.0 True",
     ]
 
-    assert repr(t[1]).splitlines() == trepr
-    assert repr(t[np.int_(1)]).splitlines() == trepr
-    assert repr(t[np.uint(1)]).splitlines() == trepr
+
+@pytest.mark.parametrize(
+    "column,expectation",
+    [
+        pytest.param("b", np.False_, id="column_present"),
+        pytest.param("x", None, id="column_not_present"),
+    ],
+)
+def test_row_get_with_default_default(column, expectation, small_table):
+    assert small_table[0].get(column) is expectation
+
+
+@pytest.mark.parametrize(
+    "column,expectation",
+    [
+        pytest.param("b", np.True_, id="column_present"),
+        pytest.param("y", False, id="column_not_present"),
+    ],
+)
+def test_row_get_with_specified_default(column, expectation, small_table):
+    assert small_table[1].get(column, False) is expectation
```

Comment:
While I understand and appreciate the idea of atomic tests, this just seems overboard. Keeping in mind "Simple is better than complex", what about a simple version that is much faster for people to read and confirm that it has the right behavior. It might also be just as fast to run in tests because there must be a substantial overhead in pytest to generate and collect 4 different tests.
```
def test_row_get():
    t = table.Table([[1, 2]], names=["a"])
    assert t[0].get("a") == 1
    assert t[0].get("x") is None
    assert t[1].get("a", False) == 1
    assert t[1].get("x", False) is False
```

Reply:
> simple version that is much faster for people to read

In my opinion well parametrized tests with detailed names make the output of `pytest` much more readable, so that the purpose of a test might even be understood without having to read the source code at all. But in this case the test is indeed very short and simple so that even if it does fail the entirety of the test code is easily readable from the traceback `pytest` prints to the terminal. Not parametrizing this test therefore does not hurt the usability of `pytest` and it does shorten the source code.